app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/reimbursement/SelectDutyTypesController.scala

```diff
@@ -0,0 +1,124 @@
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.DraftClaim.DraftC285Claim
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.JourneyStatus.FillingOutClaim
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.reimbursement.{DutyType, DutyTypesAnswer}
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.{Error, upscan => _}
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.util.toFuture
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.utils.Logging
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.{reimbursement => pages}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import scala.concurrent.{ExecutionContext, Future}
+@Singleton
+class SelectDutyTypesController @Inject() (
+  val authenticatedAction: AuthenticatedAction,
+  val sessionDataAction: SessionDataAction,
+  val sessionCache: SessionCache,
+  cc: MessagesControllerComponents,
+  val config: Configuration,
+  selectDutyTypesPage: pages.select_duty_types
+)(implicit ec: ExecutionContext, viewConfig: ViewConfig, errorHandler: ErrorHandler)
+    extends FrontendController(cc)
+    with WithAuthAndSessionDataAction
+    with Logging
+    with SessionDataExtractor
+    with SessionUpdates {
+
+  implicit val dataExtractor: DraftC285Claim => Option[DutyTypesAnswer] = _.dutyTypesSelectedAnswer
+
+  def showDutyTypes(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    withAnswers[DutyTypesAnswer] { (_, answer) =>
+      answer.fold(
+        Ok(selectDutyTypesPage(selectDutyTypesForm))
+      )(dutyType => Ok(selectDutyTypesPage(selectDutyTypesForm.fill(dutyType))))
+    }
+  }
+
+  def submitDutyTypes(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request: RequestWithSessionData[AnyContent] =>
+      withAnswers[DutyTypesAnswer] { (fillingOutClaim, answer) =>
+        selectDutyTypesForm
+          .bindFromRequest()
+          .fold(
+            formWithErrors => BadRequest(selectDutyTypesPage(formWithErrors)),
+            selectedAnswer =>
+              answer.fold {
+                updateDutyTypeAnswer(selectedAnswer, fillingOutClaim)
+              }(dutyTypesSelectedAnswer =>
+                if (selectedAnswer === dutyTypesSelectedAnswer) {
+                  Redirect(reimbursementRoutes.SelectDutyCodesController.start())
+                } else {
+                  updateDutyTypeAnswer(selectedAnswer, fillingOutClaim)
+                }
+              )
+          )
+      }
+    }
+
+  private def updateDutyTypeAnswer(
+    dutyTypesSelectedAnswer: DutyTypesAnswer,
+    fillingOutClaim: FillingOutClaim
+  )(implicit hc: HeaderCarrier, request: RequestWithSessionData[AnyContent]): Future[Result] = {
+    val newDraftClaim: DraftC285Claim   =
+      fillingOutClaim.draftClaim.fold(_.copy(dutyTypesSelectedAnswer = Some(dutyTypesSelectedAnswer)))
```

Comment:
It would be great to remove `DraftClaim` trait as no more `folding` will be required.

Lines 95 & 96 can be replaced with
```
FillingOutClaim.of(fillingOutClaim)(_.copy(dutyTypesSelectedAnswer = Some(dutyTypesSelectedAnswer)))
```

Reply:
cool, I didn't see that method - looks like you added it a while ago. Changed it accordingly.